src/main/java/ma/hibernate/dao/AbstractDao.java

```diff
@@ -1,11 +1,34 @@
 package ma.hibernate.dao;
 
+import org.hibernate.Session;
 import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
 
-public abstract class AbstractDao {
+public abstract class AbstractDao<T> {
     protected final SessionFactory factory;
 
     protected AbstractDao(SessionFactory sessionFactory) {
         this.factory = sessionFactory;
     }
+
+    T create(T entity) {
```

Comment:
You could do it inside PhoneDAO implementation

Reply:
I don't want to write the implementation manually for every model dao create method :)